cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -124,7 +125,16 @@ func newHcsTask(
 
 	owner := filepath.Base(os.Args[0])
 
-	io, err := cmd.NewNpipeIO(ctx, req.Stdin, req.Stdout, req.Stderr, req.Terminal)
+	var io cmd.UpstreamIO
+
+	u, err := url.Parse(req.Stdout)
+
+	if err == nil && u.Scheme == "binary" {
```

Comment:
We should throw an error if `u.Scheme` is anything other than `binary`. Right now we would pass the URI to `NewNpipeIO`, which would fail trying to deal with a URI.

Reply:
done